src/main/java/school/lemon/changerequest/java/extendedinteger/ExtendedInteger.java

```diff
@@ -49,7 +49,7 @@ public static ExtendedInteger parseInt(char[] value) {
 
 
     public static ExtendedInteger parseInt(String value) {
-        if (("").equals(value)) return null;
+        if ("".equals(value) || value == null) return null;
```

Comment:
`null` check should always be the first one - it's just good way to implement such check.

You implementation will be OK, but it's quite strange to check value for empty and only after that for null.